### PR TITLE
all: namespace changes, improve types + examples sync

### PR DESCRIPTION
**Pull request by ahjohannessen**
- remove direct null usage in `EsException` and
   change all es exceptions to final case classes.

 - remove usage of accessing `EsSettings.Default`
   in case classes ctor in order to elimate default
   values being tied to possible side-effects such
   as IO and randomId.

 - remove default arguments from `EsSettings` and
   `HttpSettings` in core.

 - add package `core` and make type alias in client
   project.

 - add smart constructors in package object `eventstore`
   in client project such that Akka usage still gets
   default settings behavior.

 - adjust code with regards to package changes.

 - adjust various code and specs due to removal or
   deprecation of default usage `randomId` in core.

 - add helper functions in `TestData` wrt. default
   ctor arguments being removed from types.

 - sync examples in readme

Note: The Scala client API should not break existing code due to type aliases / smart ctors in the client project's `eventstore` package object. The Java client API breaks because Java has no concept of type aliases.

### CODE REVIEW

**Comment by ahjohannessen** [comment]
@t3hnar WDYT?

File: client/src/main/scala/eventstore/akka/Settings.scala [hunk 1]
```diff
@@ -60,12 +61,11 @@ final case class Settings(
 
 object Settings {
 
-  lazy val EsDefault = EsSettings()
-  lazy val Default   = Settings(ConfigFactory.load())
+  lazy val Default = Settings(ConfigFactory.load())
```
Comment (t3hnar) [inline client/src/main/scala/eventstore/akka/Settings.scala]: in general I would ban all `ConfigFactory.load()` calls :)

File: core/src/main/scala/eventstore/core/SystemError.scala [hunk 2]
```diff
@@ -9,29 +10,28 @@ trait SystemError extends RuntimeException with NoStackTrace with Serializable
...
+@SerialVersionUID(1L) case object BadRequest       extends ServerError
 @SerialVersionUID(1L) case object NotAuthenticated extends ServerError
 
 sealed trait OperationError extends ServerError
 
 object OperationError {
-  @SerialVersionUID(1L) case object PrepareTimeout extends OperationError
-  @SerialVersionUID(1L) case object CommitTimeout extends OperationError
-  @SerialVersionUID(1L) case object ForwardTimeout extends OperationError
+  @SerialVersionUID(1L) case object PrepareTimeout       extends OperationError
```
Comment (t3hnar) [inline core/src/main/scala/eventstore/core/SystemError.scala]: what if we will drop all `@SerialVersionUID(1L)` ?

File: core/src/main/scala/eventstore/core/operations/UpdatePersistentSubscriptionInspection.scala [hunk 3]
```diff
@@ -12,9 +13,9 @@ private[eventstore] final case class UpdatePersistentSubscriptionInspection(out:
 
   def decision(error: UpdatePersistentSubscriptionError) = {
     val result = error match {
-      case AccessDenied => new AccessDeniedException(s"Write access denied for stream $streamId")
-      case DoesNotExist => new InvalidOperationException(s"Subscription group ${out.groupName} on stream $streamId does not exist")
-      case Error(msg)   => new ServerErrorException(msg.orNull)
+      case AccessDenied => AccessDeniedException(s"Write access denied for stream $streamId")
+      case DoesNotExist => InvalidOperationException(s"Subscription group ${out.groupName} on stream $streamId does not exist")
+      case Error(msg)   => ServerErrorException(msg.orNull)
```
Comment (t3hnar) [inline core/src/main/scala/eventstore/core/operations/UpdatePersistentSubscriptionInspection.scala]: are we sure about `.orNull` here ?

File: core/src/main/scala/eventstore/core/SystemError.scala (hunk 2 above)
Comment (ahjohannessen) [reply core/src/main/scala/eventstore/core/SystemError.scala]: Agree

File: client/src/main/scala/eventstore/akka/Settings.scala (hunk 1 above)
Comment (ahjohannessen) [reply client/src/main/scala/eventstore/akka/Settings.scala]: Agree :)

File: core/src/main/scala/eventstore/core/operations/UpdatePersistentSubscriptionInspection.scala (hunk 3 above)
Comment (ahjohannessen) [reply core/src/main/scala/eventstore/core/operations/UpdatePersistentSubscriptionInspection.scala]: No, could also do `.getOrElse(“”)`. Will look at this closer

**Pull request merged**